lzfse/src/lib.rs

```diff
@@ -50,12 +50,12 @@ pub enum Error {
 pub fn encode_buffer(input: &[u8], output: &mut [u8]) -> Result<usize, Error> {
     let out_size = unsafe {
         ffi::lzfse_encode_buffer(
-            output.as_ptr() as *mut _,
```

Comment:
I think this may have technically be undefined behavior, since this uses a shared reference, and mutates through it.